pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+	}
+
+	// Create install Collabora job
+	err = createInstallCollaboraJob(comp, svc)
+	if err != nil {
+		return runtime.NewWarningResult("Failed to add Collabora install Job: " + err.Error())
+	}
+
+	return runtime.NewNormalResult("Collabora deployed and configured!")
+}
+
+func AddCollaboraSts(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+
+	sts := &v1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels: map[string]string{
+				"app": comp.GetName() + "-collabora-code",
+			},
+		},
+		Spec: v1.StatefulSetSpec{
+			Replicas: ptr.To[int32](1),
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{"app": comp.GetName() + "-collabora-code"},
+			},
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   comp.GetName() + "-collabora-code",
+					Labels: map[string]string{"app": comp.GetName() + "-collabora-code"},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: comp.GetName() + "-collabora-code-sa",
+					// SecurityContext: &corev1.PodSecurityContext{
+					// 	// https://github.com/CollaboraOnline/online/blob/master/docker/from-packages/Dockerfile#L138
+					// 	RunAsUser: ptr.To[int64](100),
+					// },
+					Containers: []corev1.Container{
+						{
+							Env: []corev1.EnvVar{
+								{
+									Name:  "DONT_GEN_SSL_CERT",
+									Value: "true",
+								},
+								{
+									// https://github.com/nextcloud/all-in-one/discussions/2314
+									// https://github.com/CollaboraOnline/online/issues/3102
+									Name:  "extra_params",
+									Value: "--o:net.proto=IPv4",
+								},
+							},
+							Image: "collabora/code:24.04.8.1.1",
+							Name:  comp.GetName() + "-collabora-code",
+							Ports: []corev1.ContainerPort{
+								{
+									ContainerPort: 9980,
+								},
+							},
+							Resources: corev1.ResourceRequirements{
+								Limits: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("4"),
+									corev1.ResourceMemory: resource.MustParse("8Gi"),
+								},
+								Requests: corev1.ResourceList{
+									corev1.ResourceCPU:    resource.MustParse("500m"),
+									corev1.ResourceMemory: resource.MustParse("500Mi"),
+								},
+							},
+
+							SecurityContext: &corev1.SecurityContext{
+								Capabilities: &corev1.Capabilities{
+									Add: []corev1.Capability{
+										// https://help.nextcloud.com/t/nextcloud-office-could-not-establish-connection-to-the-collabora-online-server/185721/8
+										"FOWNER",
+										"SYS_CHROOT",
+										"CHOWN",
+										// https://github.com/chrisingenhaag/helm/blob/main/charts/collabora-code/values.yaml#L64
+										"MKNOD",
+									},
+								},
+								// https://github.com/CollaboraOnline/online/blob/master/docker/from-packages/Dockerfile#L138
+								RunAsUser: ptr.To[int64](100),
```

Comment:
How this is supposed to work in OpenShift? Doesn't OpenShift already manage this paramter?

Reply:
This is yet another hack required to make it working, please take a look here: https://github.com/CollaboraOnline/online/blob/master/docker/from-packages/Dockerfile#L138

this User setting is one of the reasons why we need separate SCC to run this image

Reply:
As long as this user works in OpenShift I am fine.